web/Dockerfile

```
@@ -1,16 +1,191 @@
 
-FROM ubuntu:focal
+FROM ubuntu:bionic
+
 # Install core components
 ENV DEBIAN_FRONTEND noninteractive
-RUN apt update -y && apt install wget sudo -y
-# This will install MISP Core
-RUN wget -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
-RUN sed -i 's/> \/dev\/tty//' /tmp/INSTALL.sh
-RUN sed -i 's/> \/dev\/null//' /tmp/INSTALL.sh 
-RUN sed -i 's/checkInstaller$/#checkInstaller/' /tmp/INSTALL.sh
-RUN bash /tmp/INSTALL.sh -c
 
-VOLUME /var/www/MISP
-EXPOSE 80 443
+RUN apt-get update && \
+    apt-get dist-upgrade -y && apt-get autoremove -y && apt-get clean && \
+    apt-get install -y software-properties-common && \
+    apt-get install -y postfix && \
+    apt-get install -y mysql-client curl gcc git gnupg-agent \
+        make python openssl redis-server sudo vim zip locales
+
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+
+# Apache
+RUN apt-get install -y apache2 apache2-doc apache2-utils && \
+    a2dismod status && \
+    a2dissite 000-default
+
+# PHP 7.2 and install MySQL PDO extension
+RUN apt-get install -y libapache2-mod-php php php-dev php-json \
+        php-mysql php-redis php-xml php-mbstring \
+        php-gd php-pear php-opcache \
+        pkg-config libbson-1.0 libmongoc-1.0-0
+
+# Fix php.ini with recommended settings
+RUN sed -i \
+        -e "s/max_execution_time = 30/max_execution_time = 300/" \
+        -e "s/memory_limit = 128M/memory_limit = 2048M/" \
+        -e "s/upload_max_filesize = 2M/upload_max_filesize = 50M/" \
+        -e "s/post_max_size = 8M/post_max_size = 50M/" \
+        /etc/php/7.2/apache2/php.ini
+
+RUN apt-get install -y python3-dev python3-pip python3-setuptools \
+        python3-lxml libjpeg-dev \
+        libxml2-dev libxslt1-dev zlib1g-dev libfuzzy-dev && \
+    apt-get install -y cron logrotate supervisor syslog-ng-core && \
+    apt-get clean
+
+WORKDIR /var/www
+RUN chown www-data:www-data /var/www
+USER www-data
+RUN git clone https://github.com/MISP/MISP.git
+WORKDIR /var/www/MISP
+RUN git checkout tags/$(git describe --tags `git rev-list --tags --max-count=1`) && \
+    git config core.filemode false && \
+    git submodule update --init --recursive && \
+    git submodule foreach --recursive git config core.filemode false
+
+USER root
+RUN pip3 install git+https://github.com/STIXProject/python-stix.git \
```

PIP is now too old which breaks the build, see question 2 here: https://pypi.org/project/opencv-python/

This should fix it, tested locally and seems to work:

```suggestion
RUN pip3 install --upgrade pip
RUN pip3 install git+https://github.com/STIXProject/python-stix.git \
```